### PR TITLE
[contracts] Fix comments, styling and small optimization for hashOrder

### PR DESCRIPTION
**Pull request by abandeali1**
## Description

- Fixes some misleading and unclear comments/styling
- Adds small optimization + readability improvement to `hashOrder`

## Checklist:

*   [ ] Prefixed the title of this PR with `[WIP]` if it is a work in progress.
*   [ ] Prefixed the title of this PR with bracketed package name(s) corresponding to the changed package(s). For example: `[sol-cov] Fixed bug`.
*   [ ] Added tests to cover my changes, or decided that tests would be too impractical.
*   [ ] Updated documentation, or decided that no doc change is needed.
*   [ ] Added new entries to the relevant CHANGELOG.jsons.

### CODE REVIEW

File: packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol [hunk 1]
```diff
@@ -123,20 +123,27 @@ contract MixinTransactions is
         bytes32 dataHash = keccak256(data);
 
         // Assembly for more efficiently computing:
-        // keccak256(abi.encode(
+        // keccak256(abi.encodePacked(
         //     EIP712_ZEROEX_TRANSACTION_SCHEMA_HASH,
         //     salt,
-        //     signerAddress,
+        //     bytes32(signerAddress),
```
Comment (Recmo) [inline packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol]: Is this a matter of style or is there a performance difference? Both should work fine.

File: packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol [hunk 2]
```diff
@@ -123,20 +123,27 @@ contract MixinTransactions is
...
+            mstore(memPtr, schemaHash)                                                               // hash of schema
+            mstore(add(memPtr, 32), salt)                                                            // salt
+            mstore(add(memPtr, 64), and(signerAddress, 0xffffffffffffffffffffffffffffffffffffffff))  // signerAddress
+            mstore(add(memPtr, 96), dataHash)                                                        // hash of data
+
+            // Compute hash
             result := keccak256(memPtr, 128)
+
+            // Increment free memory pointer
+            mstore(64, add(memPtr, 128))
```
Comment (Recmo) [inline packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol]: Wasteful. The allocated memory does not leave the scope, it should be discarded after the `keccak256` line. Increasing the `memptr` leaves it in memory, increasing max memory usages and thus gas cost.

File: packages/contracts/src/2.0.0/protocol/Exchange/libs/LibEIP712.sol [hunk 3]
```diff
@@ -59,8 +59,31 @@ contract LibEIP712 {
...
+
+            mstore(memPtr, 0x1901000000000000000000000000000000000000000000000000000000000000)  // EIP191 header
+            mstore(add(memPtr, 2), eip712DomainHash)                                            // EIP712 domain hash
+            mstore(add(memPtr, 34), hashStruct)                                                 // Hash of struct
+
+            // Compute hash
+            result := keccak256(memPtr, 66)
+
+            // Increment free memory pointer
+            mstore(64, add(memPtr, 66))
```
Comment (Recmo) [inline packages/contracts/src/2.0.0/protocol/Exchange/libs/LibEIP712.sol]: Same.

File: packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol (hunk 1 above)
Comment (abandeali1) [reply packages/contracts/src/2.0.0/protocol/Exchange/MixinTransactions.sol]: Mostly style. I just think there's less cognitive overhead in "pad everything to 32 bytes" than "ABI encode". If you already know what ABI encoding is, it's pretty obvious that you can use it to accomplish this. If you don't know what ABI encoding is, it's not obvious why we are doing it at all.

**Pull request merged**